### ISSUE TITLE
Enhancements to Macro Examples

### ISSUE THREAD

**Issue by Matejkob** [apple/swift-syntax#2169]
### Description

- [ ] Add unit tests @Matejkob #2168 
- [ ] Make implementations of all macros consistent
- [ ] Change nodes modification to use variable setters instead of `with` approach
- [ ] Add DocC tutorials for Accessor Macros
- [ ] Add DocC tutorials for Declaration Macros
- [ ] Add DocC tutorials for Expression Macros
- [ ] Add DocC tutorials for Extension Macros
- [ ] Add DocC tutorials for Member Macros
- [ ] Add DocC tutorials for MemberAttribute Macros
- [ ] Add DocC tutorials for Peer Macros

**Comment by ahoppen** [comment]
Tracked in Apple’s issue tracker as rdar://115159302

**Comment by natikgadzhi** [comment]
@Matejkob, is the link to https://github.com/apple/swift-syntax/pull/2168 intentional? Help with unit tests is very welcome — I would be happy to learn from you!

**Comment by Matejkob** [comment]
> @Matejkob, is the link to #2168 intentional? Help with unit tests is very welcome — I would be happy to learn from you!

Yes, it is intentional. I've added all missing unit tests in liked PR.
